Automattic/jetpack
# Update/rewrite use my jetpack connection hook to typescript

## Proposed changes:

* Rename getMyJetpackWindow state to specify it is the initial state
* Add 2 more util functions to get myJetpackRest and myJetpackConnection window state
* Convert useMyJetpackConnection to typescript

### Other information:


## Jetpack product discussion

P2: p1HpG7-rcF-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. Checkout this branch via the Jetpack Beta plugin or your local dev environment
2. Make sure the connection section works as expected
( If connected )
![image](https://github.com/Automattic/jetpack/assets/65001528/2ae2acac-b5fa-47f1-9749-cd103020d06a)

( if not connected )
![image](https://github.com/Automattic/jetpack/assets/65001528/fb31bfda-a595-4357-ae14-2c39a4b3f9a2)
![image](https://github.com/Automattic/jetpack/assets/65001528/b50cc982-504d-49ef-b73f-b33ec628b991)

( connection screen loads right )
![image](https://github.com/Automattic/jetpack/assets/65001528/1beecf67-aa1d-4ec0-8af5-07c7e512e292)

3. Make sure the connection flow still works
